app/experimenter/experiments/tests/test_tasks.py

```diff
@@ -373,3 +374,75 @@ def test_internal_only_does_not_add_comment(self):
         self.mock_bugzilla_requests_post.assert_not_called()
 
         self.assertEqual(Notification.objects.count(), 0)
+
+
+class TestUpdateExperimentStatus(
+    MockRequestMixin, MockNormandyMixin, TestCase
+):
+
+    def setUp(self):
+        super().setUp()
+
+        self.experiment = ExperimentFactory.create_with_status(
+            target_status=Experiment.STATUS_ACCEPTED, normandy_id=1234
+        )
+
+    def test_successful_experiment_status_update(self):
+        with MetricsMock() as mm:
```

Comment:
I think we should actually take out the statsd assertions from the tests altogether.  In general in the past I've avoided testing them, it's a huge amount of verbosity and boilerplate in the tests for very little value.  I might file a ticket to remove all the statsd test assertions.

Reply:
Out of curiosity, could you think of a better way to test this? I know that making metrics easier to test is one of @willkg's goals with Marcus. I wonder if the library could do a better job helping here?

At least for me, metrics are very important because if the metrics break, we can't tell what's going on in prod. To me that makes the tests pretty valuable, but I agree they're pretty verbose. I wonder if there is some middleground where we could still assert the right metrics are generated, but is less verbose than this?

Reply:
I test the metrics I have monitoring alerts tied to and the ones that I use to diagnose problems. The tests verify the keys and values.

Maybe you could compare the list of metrics generated against an expected list:

```
expected = [
    (markus.INCR, "experiments.tasks.update_experiment_status.started", 1, []),
    (markus.INCR, "experiments.tasks.update_experiment_status.updated", 1, []),
    (markus.INCR, "experiments.tasks.update_experiment_status.completed", 1, []),
    WHATEVER
]

self.assertEqual(mm.get_records(), expected)
```

Are they generated in the same order? What's the missing one?